### ISSUE TITLE
WFS 2.0 next/previous links include available cookies in the URL when using maxfeatures 

### ISSUE THREAD

**Issue by sdlime** [mapserver/mapserver#5629]
I noticed that the URLs built for paging for WFS 2.0 include all the params used with the original request **plus** any other params added from the list of available cookies. So if the requester has cookies set that are available to MapServer then they appear too. For example we serve a page that uses Google Analytics from the same server I'm making a WFS request. In this case I see the URL encoded __utmz value added to the WFS next and previous URLs. 

Now this is my data so it's probably not a huge issue (been like this forever) but if I didn't notice and shared the response text it would contain those data. If I just share the URL used to create the data then it's not an issue.

Solution? I guess be more explicit in constructing the next/previous URLs.

@rouault, what do you think?

--Steve

**Comment by rouault** [comment]
Fixed per https://github.com/mapserver/mapserver/commit/eabcb17b63787ad35173635132c6633def21f7f8 (arg: I got the ticket number in the commit message)

**Comment by jratike80** [comment]
I tried to read the commit but I am not sure if paging of mapfiles which utilize variable substitution will work after this change or not.

**Comment by rouault** [comment]
@sdlime I'm afraid @jratike80 is right and my change breaks variable substitution. Looking at the code for variable substitution, it looks like there's no way to get the list of parameters that are used for substitutions. I believe the safest would be to revert my change.

**Comment by sdlime** [comment]
@roualt, I agree... In terms of a fix I think avoiding any params sourced from cookies would make sense. They would be available via that same mechanism, if needed, for subsequent requests. I think we can capture the source (get, post, cookie) as the list of params is built which would make filtering for you easy.

—Steve
________________________________

**Comment by sdlime** [comment]
@rouault, did you revert?

**Comment by rouault** [comment]
I've just reverted

**Comment by sdlime** [comment]
I can take ownership of this one. I think we can alter the cgiRequestObj to track parameter source and use that to omit cookie-based parameters. I'll target 7.2.1... --Steve
